### PR TITLE
proof of concept: automatically generate subnav

### PR DESCRIPTION
**Pull request by afeld**
...rather than having to create by hand. Open questions:

- I personally think constructing tables of contents by hand is an anti-pattern. Do others like this idea of automatically generating them?
- This is a breaking change, as the `subnav` page variable is no longer supported. While , do we want to make it backwards-compatible?
- The subnav only shows up if the page has a sidenav. Open to changing that?
- Do we want to include the gem by default? Do we enable by default on all pages?

### CODE REVIEW

**Comment by thisisdano** [comment]
Jumping into this now...

**Comment by thisisdano** [comment]
> I personally think constructing tables of contents by hand is an anti-pattern. Do others like this idea of automatically generating them?

Yes — I strongly agree with this.

> This is a breaking change, as the `subnav` page variable is no longer supported. While , do we want to make it backwards-compatible?

Does this mean to continue supporting the `subnav` variable and font matter section? With some kind of conditional in the sidenav template? If so, I'd be inclined to just break it along with the breaking changes in 4.0.0 (along with USWDS 2.x support). 

THAT SAID... on the flip side, if we continue to support the `subnav` front matter, it would not only provide backward compatibility, but also provide a way to override and/or customize the sidenav — I could imagine an instance where the user might want to use a shortened version of the heading name to make it more readable.

So maybe backward compatibility wins the day here?

> The subnav only shows up if the page has a sidenav. Open to changing that?

I _think_ so — I'll need to play around with this for a bit. 

> Do we want to include the gem by default? Do we enable by default on all pages?

That is, something like generate by default unless `toc: false` in the font matter? That seems reasonable.

**Comment by thisisdano** [comment]
> The subnav only shows up if the page has a sidenav. Open to changing that?

Can we talk a bit more about what this means? So in the case that there's no sidenav — like, it's a single-page section — what would we see in the sidenav area? Something like the following (in the current sidenav style)?

```
page title
├── subnav-item
│   ├── subnav-child-item
│   ├── subnav-child-item
├── subnav-item
...
```
Or some new kind of sidenav?

The current sidenav setup (a referenced YAML key in a data file, with manual setup?) feels a bit janky, too, in the same way constructing TOCs is an antipattern. We don't need to solve this now, but once we start talking about autogenerating saidenavs it starts to beg the question.

That said, having TOCs on long structured pages is good usability. I would be inclined to add a sidenav to docs pages. But are there any instances where the `post` layout would be used on non-"docs" content? That is, does a blog post demand a sidebar and TOC? Arguably not.

Perhaps there should be a way to call out a page as a documentation-style page, independent of specifically adding a unique `sidenav` attribute in the front matter. Something like `layout: docs`?

**Comment by thisisdano** [comment]
TLDR: I like this change, support backward compatibility, and feel that that's the best scope for the current work. Autogenerating a complete (or standalone) sidenav or creating new layout types could be different features.

**Comment by adborden** [comment]
Hey folks, the @18F/tts-tech-portfolio is putting this repository into "maintenance mode" and will only accept fixes for the time being. Tech Portfolio and TTS Digital Council are working on a strategy for TTS microsites which will inform the future of this theme.

Rather than leave open pull requests to go stale, we are respectfully going to close this PR. You can [provide any feedback by opening a new issue](https://github.com/18F/uswds-jekyll/issues/new).
